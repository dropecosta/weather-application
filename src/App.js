import { useState } from "react";
import { WEATHER_API_URL, WEATHER_API_KEY } from "./api/api";
import { AppContext } from "./context/context";
import Header from "./components/Header/Header";
import Search from "./components/Search/Search";
import CurrentWeather from "./components/CurrentWeather/CurrentWeather";
import Forecast from "./components/Forecast/Forecast";
import "./App.scss";

const App = () => {
  const [currentCelciusWeather, setCurrentCelciusWeather] = useState(null);
  const [forecastCelciusWeather, setForecast] = useState(null);
  const [currentFarenheithWeather, setCurrentFarenheithWeather] = useState(null);
  const [forecastFarenheithWeather, setForecastFarenheithWeather] = useState(null);
  const [toggled, setToggled] = useState(false);

  const handleOnSearchChange = (searchData) => {
    const [lat, lon] = searchData.value.split(" ");

    const currentWeatherCelsiusFetch = fetch(`${WEATHER_API_URL}/weather?lat=${lat}&lon=${lon}&appid=${WEATHER_API_KEY}&units=metric`);
    const forecastCelciusFetch = fetch(`${WEATHER_API_URL}/forecast?lat=${lat}&lon=${lon}&appid=${WEATHER_API_KEY}&units=metric`);
    const currentWeatherFarenheithFetch = fetch(`${WEATHER_API_URL}/weather?lat=${lat}&lon=${lon}&appid=${WEATHER_API_KEY}&units=imperial`);
    const forecastFarenheithFetch = fetch(`${WEATHER_API_URL}/forecast?lat=${lat}&lon=${lon}&appid=${WEATHER_API_KEY}&units=imperial`);

    Promise.all([currentWeatherCelsiusFetch, forecastCelciusFetch, currentWeatherFarenheithFetch, forecastFarenheithFetch])
      .then(async (response) => {
        const weatherCelciusResponse = await response[0].json();
        const forcastCelciusResponse = await response[1].json();
        const weatherFarenheithResponse = await response[2].json();
        const forcastFarenheithResponse = await response[3].json();

        setCurrentCelciusWeather({ city: searchData.label, ...weatherCelciusResponse });
        setForecast({ city: searchData.label, ...forcastCelciusResponse });
        setCurrentFarenheithWeather({ city: searchData.label, ...weatherFarenheithResponse });
        setForecastFarenheithWeather({ city: searchData.label, ...forcastFarenheithResponse });
      })
      .catch(error => console.log(error));
  };

  function handleChange(toggled) {
    setToggled(toggled);
  }

  return (
    <AppContext.Provider value={{ currentCelciusWeather, forecastCelciusWeather, currentFarenheithWeather, forecastFarenheithWeather}}>
      <div className='app'>
        <div className="container">
          <Header />
          <Search onSearchChange={handleOnSearchChange} />
          {currentCelciusWeather && ( <CurrentWeather onChange={handleChange} />)}
          {forecastCelciusWeather && <Forecast onChange={toggled} />}
        </div>
      </div>
    </AppContext.Provider>
  );
};

export default App;
