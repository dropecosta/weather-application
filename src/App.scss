.app {
    margin: 0;
    height: 100vh;
    width: 100%;

  .container {
    max-width: 1080px;
    margin: 20px auto;
    padding-bottom: 30px;
  }
  
}