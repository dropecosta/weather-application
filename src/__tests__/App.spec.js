import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "../App";

describe("App component tests", () => {
  it("should render APP component with container class", () => {
    const { container } = render(<App />);
    expect(container.firstChild).toHaveClass("app");
  });

  it("should match the snapshot", () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
    