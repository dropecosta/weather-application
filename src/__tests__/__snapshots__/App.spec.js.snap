// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests should match the snapshot 1`] = `
<div
  className="app"
>
  <div
    className="container"
  >
    <div
      className="header"
    >
      <h2>
        Weather APP
      </h2>
      <div
        className="toggle-container"
      >
        <img
          alt="moon"
          className="darkmode-icon-moon"
          src="moon.svg"
        />
        <input
          checked={false}
          className="switch-checkbox"
          id="switch-darkmode"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="switch-label"
          htmlFor="switch-darkmode"
          style={
            Object {
              "background": false,
            }
          }
        >
          <span
            className="switch-button"
          />
        </label>
        <img
          alt="sun"
          className="darkmode-icon-sun"
          src="sun.svg"
        />
      </div>
    </div>
    <div
      className="search"
    >
      <div
        className=" css-b62m3t-container"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-1s2u09g-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-319lph-ValueContainer"
          >
            <div
              className=" css-14el2xx-placeholder"
              id="react-select-3-placeholder"
            >
              Search for city
            </div>
            <div
              className=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-3-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-3-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-tj5bde-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
